### PR TITLE
Show rate game button on game end

### LINKED ISSUE #729: Test if players are asked to rate game when it finishes

### PR DESCRIPTION
**Pull request by 1-alex98**
Fixes #729

### CODE REVIEW

File: src/main/java/com/faforever/client/game/GameService.java [hunk 1]
```diff
@@ -198,12 +204,12 @@ public GameService(ClientProperties clientProperties, FafService fafService,
     JavaFxUtil.attachListToMap(games, uidToGameInfoBean);
   }
 
-  
+
```
Comment (micheljung) [inline src/main/java/com/faforever/client/game/GameService.java]: no idea where all these empty lines come from, but they should be removed.

**Review by micheljung — CHANGES_REQUESTED** [review]

File: src/test/java/com/faforever/client/replay/OnlineReplayVaultControllerTest.java [hunk 2]
```diff
@@ -129,7 +137,14 @@ public void testOnSearchButtonClicked() throws Exception {
   }
 
   @Test
-  public void testOnSearchButtonClickedHandlesException() throws Exception {
+  public void testShowReplayEvent() {
+    Replay replay = new Replay();
+    instance.onDisplay(new ShowReplayEvent(replay));
+    verify(replayDetailController).setReplay(replay);
```
Comment (micheljung) [inline src/test/java/com/faforever/client/replay/OnlineReplayVaultControllerTest.java:144]: Can we test the ChangeListener as well?

File: src/main/java/com/faforever/client/game/GameService.java [hunk 3]
```diff
@@ -465,12 +471,23 @@ private void startGame(GameLaunchMessage gameLaunchMessage, Faction faction, Rat
...
-          singletonList(new Action(i18n.get("game.rate"), actionEvent -> replayService.findById(currentGameId)
-              .thenAccept(replay -> externalReplayInfoGenerator.showExternalReplayInfo(replay, String.valueOf(currentGameId)))))));
+          singletonList(new Action(i18n.get("game.rate"), actionEvent -> replayService.findById(id)
+              .thenAccept(replay -> Platform.runLater(() -> {
+                if (replay.isPresent()) {
+                  eventBus.post(new ShowReplayEvent(replay.get()));
+                } else {
+                  notificationService.addNotification(new ImmediateNotification(i18n.get("replay.notFoundTitle"), i18n.get("replay.replayNotFoundText", id), Severity.WARN));
+                }
+              }))))));
```
Comment (micheljung) [inline src/main/java/com/faforever/client/game/GameService.java:481]: Is there a unit test for this as well?

**Review by micheljung — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/game/GameService.java (hunk 1 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/game/GameService.java]: They are not empty the ide only removed the whitespsaces...

Comment (micheljung) [reply src/main/java/com/faforever/client/game/GameService.java]: I checked in my IDE and they are.
If they weren't, how could the whitespace have been removed? ;)

File: src/main/java/com/faforever/client/game/GameService.java (hunk 3 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/game/GameService.java:481]: now yes

**Comment by 1-alex98** [comment]
@micheljung I need that for the discord thing

File: src/test/java/com/faforever/client/replay/OnlineReplayVaultControllerTest.java (hunk 2 above)
Comment (1-alex98) [reply src/test/java/com/faforever/client/replay/OnlineReplayVaultControllerTest.java:144]: Did that

File: src/main/java/com/faforever/client/preferences/NotificationsPrefs.java [hunk 4]
```diff
@@ -37,7 +38,7 @@ public NotificationsPrefs() {
     infoSoundEnabled = new SimpleBooleanProperty(true);
     warnSoundEnabled = new SimpleBooleanProperty(true);
     errorSoundEnabled = new SimpleBooleanProperty(true);
-    transientNotificationsEnabled = new SimpleBooleanProperty(true);
+    eventNotificationsEnabled = new SimpleBooleanProperty(true);
```
Comment (micheljung) [inline src/main/java/com/faforever/client/preferences/NotificationsPrefs.java]: Why did you rename this? Semantically, this is a step backwards

File: src/main/java/com/faforever/client/preferences/NotificationsPrefs.java [hunk 5]
```diff
@@ -67,16 +69,16 @@ public BooleanProperty soundsEnabledProperty() {
     return soundsEnabled;
   }
 
-  public boolean isTransientNotificationsEnabled() {
-    return transientNotificationsEnabled.get();
+  public boolean getEventNotificationsEnabled() {
```
Comment (micheljung) [inline src/main/java/com/faforever/client/preferences/NotificationsPrefs.java]: This shouldn't be `get` but `is`, as it was before

File: src/main/resources/i18n/messages.properties [hunk 6]
```diff
@@ -632,6 +632,10 @@ ranked1v1.showMaps=Show ladder maps
 game.create.onlyFriends=Only show to friends
 settings.language.helpUs.text=Downlord's FAF Client's localization is a community effort. If you want to see the client translated into another language we'd love your help.
 settings.language.helpUs.button=Help us translate
+replay.notFoundTitle=Oh oh. No replay for this found.
```
Comment (micheljung) [inline src/main/resources/i18n/messages.properties]: I'm sure we can come up with a better title than this. Also, titles don't end with periods.

**Review by micheljung — CHANGES_REQUESTED** [review]

**Review by micheljung — APPROVED** [review]
